core/src/main/java/org/lflang/diagram/synthesis/LinguaFrancaSynthesis.java

```diff
@@ -1712,8 +1708,18 @@ private KNode addErrorComment(KNode node, String message) {
   }
 
   private Iterable<KNode> createUserComments(EObject element, KNode targetNode) {
-    if (getBooleanValue(SHOW_USER_LABELS)) {
-      String commentText = AttributeUtils.getLabel(element);
+    if (getBooleanValue(SHOW_REACTION_NAMES)) {
+      String commentText = null;
+      // If the element is a reaction, try to read its name
+      if (element instanceof Reaction) {
+        commentText = ((Reaction) element).getName();
+      }
+      // If the element does not have a name, check if it has an @label attribute
```

Comment:
I think we should still be able to write @label("Some description") next to a reactor class and show it. I think that this change should cement that for reactions, either the reaction name (if it is specified) or a numeral is shown inside the chevron, but I'm starting to think that we should still keep the `@label` around (also for reactions) to serve as additional explanation in the diagram, just as we have them for reactor classes.